Packet++/src/VrrpLayer.cpp

```diff
@@ -0,0 +1,533 @@
+#define LOG_MODULE PacketLogModuleVrrpLayer
+
+#include <SystemUtils.h>
+#include "PacketUtils.h"
+#include "Logger.h"
+#include "EndianPortable.h"
+#include "IPv4Layer.h"
+#include "PayloadLayer.h"
+#include "VrrpLayer.h"
+
+namespace pcpp {
+
+	/*************
+	 * VrrpLayer
+	 *************/
+
+	VrrpLayer::VrrpLayer(ProtocolType subProtocol) {
+		m_DataLen = VRRP_PACKET_FIX_LEN;
+		m_Data = new uint8_t[m_DataLen];
+		memset(m_Data, 0, m_DataLen);
+		m_Protocol = subProtocol;
+
+		getPacketPtr()->ipAddrCount = 0;
+		getPacketPtr()->version = getVersionFromData();
+
+		m_addressType = IPAddress::IPv4AddressType;
+	}
+
+	bool VrrpLayer::isDataValid(size_t dataLen) {
+		return (dataLen > VRRP_PACKET_FIX_LEN);
+	}
```

Comment:
1. I think this method should accept a pointer to the data and length: `VrrpLayer::isDataValid(uint8_t* data, size_t dataLen)` and then check: `data && dataLen > ...`
2. The minimum data length to parse a packet is `sizeof(vrrp_packet)`, no?

Reply:
Yes, you are right. I have add function 'getVersionFromData' to change the parse action.